acceptance-tests/src/test/java/io/blueocean/ath/offline/personalization/FavoritesNavigationTest.java

```diff
@@ -0,0 +1,221 @@
+package io.blueocean.ath.offline.personalization;
+
+import io.blueocean.ath.ATHJUnitRunner;
+import io.blueocean.ath.BaseUrl;
+import io.blueocean.ath.GitRepositoryRule;
+import io.blueocean.ath.Login;
+import io.blueocean.ath.ResourceResolver;
+import io.blueocean.ath.WebDriverMixin;
+import io.blueocean.ath.api.classic.ClassicJobApi;
+import io.blueocean.ath.factory.ActivityPageFactory;
+import io.blueocean.ath.factory.ClassicPipelineFactory;
+import io.blueocean.ath.factory.FreestyleJobFactory;
+import io.blueocean.ath.factory.MultiBranchPipelineFactory;
+import io.blueocean.ath.factory.RunDetailsPipelinePageFactory;
+import io.blueocean.ath.model.AbstractPipeline;
+import io.blueocean.ath.model.ClassicPipeline;
+import io.blueocean.ath.model.Folder;
+import io.blueocean.ath.model.FreestyleJob;
+import io.blueocean.ath.model.MultiBranchPipeline;
+import io.blueocean.ath.pages.blue.ActivityPage;
+import io.blueocean.ath.pages.blue.BranchPage;
+import io.blueocean.ath.pages.blue.FavoritesDashboardPage;
+import io.blueocean.ath.sse.SSEClientRule;
+import io.blueocean.ath.sse.SSEEvents;
+import io.jenkins.blueocean.util.HttpRequest;
+import org.apache.log4j.Logger;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+/**
+ * @author cliffmeyers
+ */
+@Login
+@RunWith(ATHJUnitRunner.class)
+public class FavoritesNavigationTest implements WebDriverMixin {
+    private static final Logger logger = Logger.getLogger(FavoritesNavigationTest.class);
+    private static final Folder FOLDER = new Folder("personalization-folder");
+
+    @Rule @Inject
+    public GitRepositoryRule git;
+
+    @Inject @Rule
+    public SSEClientRule sseClient;
+
+    @Inject
+    FavoritesDashboardPage dashboardPage;
+
+    @Inject
+    ActivityPageFactory activityPageFactory;
+
+    @Inject
+    RunDetailsPipelinePageFactory runDetailsPageFactory;
+
+    @Inject
+    ClassicJobApi jobApi;
+
+    @Inject
+    FreestyleJobFactory freestyleFactory;
+
+    @Inject
+    ClassicPipelineFactory pipelineFactory;
+
+    @Inject
+    MultiBranchPipelineFactory multibranchFactory;
+
+    @Inject @BaseUrl
+    String base;
+
+    private ResourceResolver resources;
+
+    private HttpRequest httpRequest() {
+        return new HttpRequest(base + "/blue/rest");
+    }
+
+    @Before
+    public void setUp() throws IOException {
+        resources = new ResourceResolver(getClass());
+
+        String user = "alice";
+        logger.info(String.format("deleting any existing favorites for %s", user));
+
+        httpRequest()
+            .Delete("/users/{user}/favorites/")
+            .urlPart("user", user)
+            .auth(user, user)
+            .status(204)
+            .as(Void.class);
+    }
```

Comment:
There is some boilerplate in the test classes that I want to refactor into a superclass.